app/(gcforms)/[locale]/(form administration)/form-builder/components/dialogs/DynamicRowDialog/DynamicRowDialog.tsx

```diff
@@ -0,0 +1,264 @@
+"use client";
+import React, { useCallback, useEffect, useState } from "react";
+import { useTranslation } from "@i18n/client";
+import { Button, Alert } from "@clientComponents/globals";
+import { useTemplateStore } from "@lib/store/useTemplateStore";
+import { Dialog, useDialogRef } from "@formBuilder/components/shared/Dialog";
+import { CustomEventDetails, EventKeys, useCustomEvent } from "@lib/hooks/useCustomEvent";
+import { FormElementWithIndex } from "@lib/types/form-builder-types";
+import { dynamicRowType } from "@lib/types/form-types";
+
+export const TextInput = ({ label, children }: { label: string; children: React.ReactElement }) => {
+  return (
+    <div className="mb-4 flex rounded-md border-1 border-black">
+      <label className="block rounded-l-md border-r-1 border-black bg-slate-50 p-4 text-sm">
+        {label}
+      </label>
+      {React.cloneElement(children, {
+        className: "block w-full rounded-r-md p-2 outline-offset-[-5px]",
+      })}
+    </div>
+  );
+};
+
+export const DynamicRowDialog = () => {
+  const dialog = useDialogRef();
+  const { Event } = useCustomEvent();
+  const { t } = useTranslation("form-builder");
+
+  const [item, setItem] = useState<FormElementWithIndex | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
+
+  const [error, setError] = useState<boolean | null>(null);
+
+  const { updateField, setChangeKey } = useTemplateStore((s) => ({
+    elements: s.form.elements,
+    updateField: s.updateField,
+    setChangeKey: s.setChangeKey,
+  }));
+
+  const handleClose = () => {
+    dialog.current?.close();
+    setIsOpen(false);
+  };
+
+  const forceRefresh = () => {
```

Comment:
re-tested - definitely needed. Since the `Customize repeating set` button is responsible for passing the item to the Dialog, the template store needs to be refreshed for the next time you pop the dialog.